.gitlab-ci.yml

```diff
@@ -1,68 +1,157 @@
+
 stages:
-  - generator
-  - test-basic
-  - test-required
-  - documentation
-  - test-additional-v1.5
-  - test-additional-v1.6
+  - test # this checks the viability of the code
+  - docdata # this builds stuff and assets for the documentation
+  - documentation # this builds and deploys the actual documentation
+  - test-compat # this runs many additional compatibility tests
 
 variables:
-    GIT_STRATEGY: clone
-    DOCKER_DRIVER: overlay2
-    DOCKER_TLS_CERTDIR: ""
+  GIT_STRATEGY: clone
+  DOCKER_DRIVER: overlay2
+  DOCKER_TLS_CERTDIR: ""
+
+#
+# Predefined conditions for triggering jobs
+#
+
+.global_on_check_incoming_code: &global_on_check_incoming_code
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "external_pull_request_event"
 
-.global_settings: &global_settings_pull
+.global_on_build_doc: &global_on_build_doc
```

Comment:
not sure about the naming here ... what is up with this `on_` ?

Reply:
hm yeah, that might be a leftover from the previous scheme ("on a pullrequest")... will rethink

Reply:
replaced with `_trigger_`